app/jobs/importing/run_daily_imports_job.rb

```diff
@@ -251,5 +253,31 @@ def sync_with_cas
       GrdaWarehouse::Tasks::PushClientsToCas.new.sync!
       @notifier.ping('Pushed Clients to CAS')
     end
+
+    # UNTESTED partial implementation. This should probably go in a utility class somewhere else
+    def collect_postgres_stats
+      Dir.mktmpdir do |dir|
+        GrdaWarehouse::PostgresInspector.inspect_each do |inspector|
+          [
+            :database_stats,
+            :table_stats,
+            :index_stats,
+          ].each do |stats_method|
+            results = inspector.send(stats_method)
+            filename = Pathname.new(dir).join("#{inspector.name}.#{stats_method}.csv")
+            CSV.open(filename.to_s, 'w') do |csv|
+              headers = results.first.keys
+              csv << headers
+
+              results.each do |result|
+                csv << result.values_at(**headers)
+              end
+            end
+          end
+        end
+        raise 'implementation tbd'
+        # send_directory_to_s3(dir)
+      end
+    end
```

Comment:
1. I'd like to see the logic moved into a method on `PostgresInspector` so you could call something like `PostgresInspector.collect_statistics!`.
2. I think we should stick it in `lib/tasks/grda_warehouse.rake#hourly` and implement it as we have some others, where it runs in a default hour, but can be overridden with an ENV or site config (I'm guessing ENV is fine.)

Depending on how long it takes to run, we may want to setup it's own scheduled task or throw it into a delayed job so we don't block the `hourly` task.

Reply:
> Depending on how long it takes to run, we may want to setup it's own scheduled task or throw it into a delayed job so we don't block the hourly task.

Do we really want an **hourly** task for this? That would generate an awful lot of reports. The PT issue specified weekly. 

For our purposes of initial discovery of low hanging fruit, I suspect we could just run this manually on a few installations. I've left this out of the cron  config for now

Reply:
Sorry, I didn't mean to imply that would run every hour, only that if we add it to the hourly rake task, then we can set a `run_hour` on the remote credential, similar to how we do it here: `GrdaWarehouse::CustomImports::Config`.  It checks each hour if it should be run, and does nothing if the hour isn't correct, but queues the delayed job if the hour matches.  That way we can stagger them across installations and adjust timing easily, by updating the DB (or an ENV) with the appropriate hour.  

It looks like `remote_credentials` doesn't have an hour column, so that may be worth adding.  Running manually is probably a good idea anyway.